File: Data Structures & Algorithms/duplicate-integer/submission-1.java
class Solution {
    public boolean hasDuplicate(int[] nums) {

        int i = 0, j = 0;
        boolean bFlag = false;

        for ( i = 0; i < nums.length; i++) {
            for ( j = (i + 1); j < nums.length; j++)
            {
                if ( nums[i] == nums[j] ) {
                    bFlag = true;
                }
            }
        }
        
        return bFlag;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        // Count frequencies
        unordered_map<int,int> freq;
        for (int x : nums) freq[x]++;
        
        // Move to vector of pairs (num, count)
        vector<pair<int,int>> v(freq.begin(), freq.end());
        
        // Sort by frequency descending
        sort(v.begin(), v.end(), [](const pair<int,int>& a, const pair<int,int>& b){
            if (a.second != b.second) return a.second > b.second;
            return a.first < b.first; // tie-breaker (optional)
        });
        
        // Take top k numbers
        vector<int> ans;
        for (int i = 0; i < k; ++i) ans.push_back(v[i].first);
        return ans;
    }
};